test/opt/copy_prop_array_test.cpp

```diff
@@ -1884,6 +1884,65 @@ OpFunctionEnd
   SetTargetEnv(SPV_ENV_UNIVERSAL_1_4);
   SinglePassRunAndMatch<CopyPropagateArrays>(before, false);
 }
+
+// As per SPIRV spec, struct cannot be indexed with non-constant indices
+// through OpAccessChain, only arrays.
+// The copy-propagate-array pass tries to remove superfluous copies when the
+// original array could be indexed instead of the copy.
+//
+// This test verifies we handle this case:
+//  struct SRC { int field1; ...; int fieldN }
+//  int tmp_arr[2] = { SRC.field1, ..., SRC.fieldN }
```

Comment:
The size of the array should be <n>, not 2.

Reply:
ooops, thanks